Source/ARTRealtime.m

```diff
@@ -641,8 +641,10 @@ - (void)onConnected:(ARTProtocolMessage *)message {
             [self.connection setMaxMessageSize:message.connectionDetails.maxMessageSize];
             if (!_resuming) {
                 [self.connection setSerial:message.connectionSerial];
-                [self.logger debug:@"RT:%p msgSerial of connection \"%@\" has been reset", self, self.connection.id_nosync];
-                self.msgSerial = 0;
+                if (message.error != nil || self.options.recover == nil || [[self.options.recover stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceAndNewlineCharacterSet]] isEqualToString:@""]) {
+                    [self.logger debug:@"RT:%p msgSerial of connection \"%@\" has been reset", self, self.connection.id_nosync];
+                    self.msgSerial = 0;
```

Comment:
I get the recover bit, but why the error part i.e. if there is an error, we don't reset the message serial.  I don't really know what that is to be honest.  @ricardopereira @SimonWoolf?

Reply:
I get that you're trying to enumerate the scenarios in which you believe there should be a new connection per RTN15c3. But it's overcomplicated and possibly error-prone compared to just checking whether the `connectionId` is new (ie not the same as a previous connectionId, or there was no previous connectionId). See [the logic in ably-js](https://github.com/ably/ably-js/blob/33f4b69a50ca7b1dc25241ac7780908fdd2ed78d/common/lib/transport/connectionmanager.js#L736-L740).

Reply:
@SimonWoolf Yes, it's way simpler to just check the `connectionId`. Thanks, I will change the code.

Reply:
Fixed with 6334568 and 8cd4d50.